nanome_docking/autodock4/calculations.py

```diff
@@ -1,320 +1,175 @@
-
-    def _check_preparation(self):
-        return self._lig_process.poll() != None or self._rec_process.poll() != None
-
-    def _preparation_finished(self):
-        end = timer()
-        nanome.util.Logs.debug("Prepared ligand and receptor in", end - self._start_timer, "seconds")
-
-        if self._check_process_error(self._lig_process) or self._check_process_error(self._rec_process):
-            return
-
-        self._running = False
-        self._preparation_pending = False
-        self._parameters_preparation_pending = True
-
-    # Preparation of gpf and dpf files
-
-    def _start_parameters_preparation(self):
-        # Awful situation here
-        print("protein input converted name:", self._protein_input_converted.name)
-        grid_args = ['python2', os.path.join(os.path.dirname(__file__), 'prepare_gpf4.py'), '-l', self._ligands_input_converted.name, '-r', self._protein_input_converted.name, '-o', self._autogrid_input.name]
-        dock_args = ['python2', os.path.join(os.path.dirname(__file__), 'prepare_dpf42.py'), '-l', self._ligands_input_converted.name, '-r', self._protein_input_converted.name, '-o', self._autodock_input.name]
-
-        nanome.util.Logs.debug("Prepare grid and docking parameter files")
-        self._start_timer = timer()
-        self._grid_process = subprocess.Popen(grid_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=self.temp_dir.name)
-        self._dock_process = subprocess.Popen(dock_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=self.temp_dir.name)
-        self._running = True
-
-    def _check_parameters_preparation(self):
-        return self._grid_process.poll() != None or self._dock_process.poll() != None
-
-    def _parameters_preparation_finished(self):
-        end = timer()
-        nanome.util.Logs.debug("Prepared grid and docking parameter files in", end - self._start_timer, "seconds")
-
-        if self._check_process_error(self._grid_process) or self._check_process_error(self._dock_process):
-            return
-
-        f = open(self._autodock_input.name, 'a')
-        f.write("write_all\n")
-        f.close()
+        receptor = receptor
 
-        self._running = False
-        self._parameters_preparation_pending = False
-        self._grid_pending = True
+        docked_ligands = None
+        with tempfile.TemporaryDirectory() as self.temp_dir:
+            combined_ligands = ComplexUtils.combine_ligands(receptor, ligands)
 
-    # Autogrid
+            # Save all input files
+            receptor_file_pdb = tempfile.NamedTemporaryFile(delete=False, suffix=".pdb", dir=self.temp_dir)
+            ligands_file_pdb = tempfile.NamedTemporaryFile(delete=False, suffix=".pdb", dir=self.temp_dir)
+            receptor.io.to_pdb(receptor_file_pdb.name, pdb_options)
+            combined_ligands.io.to_pdb(ligands_file_pdb.name, pdb_options)
+            assert open(receptor_file_pdb.name).read()
+            assert open(ligands_file_pdb.name).read()
```

Comment:
What're these `assert`s for? Also is the file handle closed after this?

Reply:
The asserts were for debugging, I can remove them.

This whole function happens inside `with tempfile.TemporaryDirectory() as self.temp_dir:`, which handles all the file closing/deletions when we leave the with statement.